ffi/gn/build/rust_wasm/BUILD.gn

```diff
@@ -0,0 +1,67 @@
+# This file is part of ICU4X. For terms of use, please see the file
+# called LICENSE at the top level of the ICU4X source tree
+# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+toolchain("rust_wasm") {
+  tool("rust_bin") {
+    depfile = "{{target_out_dir}}/{{crate_name}}.d"
+    outfile = "{{target_out_dir}}/{{crate_name}}"
+    command = "rustc --crate-name {{crate_name}} {{source}} --crate-type {{crate_type}} --emit=dep-info=$depfile,link -Z dep-info-omit-d-target {{rustflags}} -o $outfile {{rustdeps}} {{externs}}"
+    command += " --target=wasm32-wasi"
+    description = "RUST $outfile"
+    outputs = [ outfile ]
+  }
+
+  tool("rust_staticlib") {
+    depfile = "{{target_out_dir}}/{{crate_name}}.d"
+    outfile = "{{target_out_dir}}/{{crate_name}}.a"
+    command = "rustc --crate-name {{crate_name}} {{source}} --crate-type {{crate_type}} --emit=dep-info=$depfile,link -Z dep-info-omit-d-target {{rustflags}} -o $outfile {{rustdeps}} {{externs}}"
+    command += " --target=wasm32-wasi"
+    description = "RUST $outfile"
+    outputs = [ outfile ]
+  }
+
+  tool("rust_rlib") {
+    depfile = "{{target_out_dir}}/{{crate_name}}.d"
+    outfile = "{{target_out_dir}}/lib{{crate_name}}.rlib"
+    command = "rustc --crate-name {{crate_name}} {{source}} --crate-type {{crate_type}} --emit=dep-info=$depfile,link -Z dep-info-omit-d-target {{rustflags}} -o $outfile {{rustdeps}} {{externs}}"
+    command += " --target=wasm32-wasi"
+    description = "RUST $outfile"
+    outputs = [ outfile ]
+  }
+
+  tool("rust_cdylib") {
+    depfile = "{{target_out_dir}}/{{crate_name}}.d"
+    outfile = "{{target_out_dir}}/lib{{crate_name}}.so"
+    command = "rustc --crate-name {{crate_name}} {{source}} --crate-type {{crate_type}} --emit=dep-info=$depfile,link -Z dep-info-omit-d-target {{rustflags}} -o $outfile {{rustdeps}} {{externs}}"
+    command += " --target=wasm32-wasi"
+    description = "RUST $outfile"
+    outputs = [ outfile ]
+  }
+
+  tool("rust_macro") {
```

Comment:
I think we shouldn't need `rust_macro` on non-host targets, but GN complains if I remove this